lib/secrets_manager_provider/configuration.ex

```diff
@@ -3,21 +3,27 @@ defmodule SecretsManagerProvider.Configuration do
   Struct for managing configurations like :client and :parser.
   """
 
+  @type args :: [{atom(), any()}]
   @type t :: %__MODULE__{
           client: module(),
+          name: String.t(),
           parser: module()
         }
 
   defstruct client: SecretsManagerProvider.ExAwsClient,
+            name: nil,
             parser: Toml
 
   @doc """
   Returns a new `Configuration` struct with defaults.
   """
-  @spec new :: t()
-  def new do
-    fields = Application.get_all_env(:secrets_manager_provider)
-
-    struct(__MODULE__, fields)
+  @spec new(args()) :: t()
+  def new(args) do
+    fields = Enum.into(args, %{}, &maybe_get_env/1)
+    struct!(__MODULE__, fields)
   end
+
+  @spec maybe_get_env(args()) :: args()
+  def maybe_get_env({name, {:system, env}}), do: {name, System.get_env(env)}
```

Comment:
I think we should match on `:name`, instead of `name` here. I can see a scenario where someone might try to set `:client` from an environment variable and the Provider would fail because `configuration."string_value".get_secrets()` would fail.

Reply:
Yes, didn't though about the modules. 🤦